assembly/src/assembler/mod.rs

```diff
@@ -132,6 +132,7 @@ impl Assembler {
 
         // compile the program
         let mut context = AssemblyContext::new(AssemblyContextType::Program);
+
         let program_root = self.compile_in_context(&program, &mut context)?;
```

Comment:
minor nit: we probably don't need this new blank line here.

Also, when we create the context for `ProgramAst`, we probably should pass the import info to it as well. Otherwise, imports won't be recorded for `ModuleContext` created for the program itself. This probably will require changing the constructor for `AssemblyContext` (or maybe create a new constructor specifically for contexts created for executable modules).